### PR TITLE
ASan parser enhancements

### PR DESCRIPTION
**Pull request by tysmith**
This resolves issue #545 and issues #546

### CODE REVIEW

**Comment by jschwartzentruber** [comment]
Do you have examples of the other param overlap logs for tests?

File: FTB/Signatures/tests/test_CrashInfo.py [hunk 1]
```diff
@@ -834,6 +839,21 @@ def test_ASanParserTestFailedAlloc():
...
+
+    crashInfo = ASanCrashInfo([], asanTraceAllocSize.splitlines(), config)
+    assert len(crashInfo.backtrace) == 1
+    assert crashInfo.backtrace[0] == "malloc"
+
+    assert crashInfo.crashAddress is None
+    assert not crashInfo.registers
+
+    assert (("AddressSanitizer: requested allocation size exceeds maximum"
+             " supported size of 0x10000000000 [@ malloc]") ==
```
Comment (jschwartzentruber) [inline FTB/Signatures/tests/test_CrashInfo.py]: What about stripping `of 0x10000000000` too?

**Comment by tysmith** [comment]
> Do you have examples of the other param overlap logs for tests?
I can find some and add them.

File: FTB/Signatures/CrashInfo.py [hunk 2]
```diff
@@ -534,7 +534,8 @@ def __init__(self, stdout, stderr, configuration, crashData=None):
                                        |negative-size-param:\s*\(size=-\d+\)
                                        |not\smalloc\(\)-ed:     # Used in case of a wild free (on unallocated memory)
                                        |not\sowned:             # Used when calling __asan_get_allocated_size() on a pointer that isn't owned
-                                       |memcpy-param-overlap:\smemory\sranges\s\[) # Bad memcpy
+                                       |\w+-param-overlap:      # Bad memcpy/strcpy/strcat... etc
```
Comment (jschwartzentruber) [inline FTB/Signatures/CrashInfo.py]: Can you add tests for these other crashes? (or at least TODO if you don't have logs handy)

Comment (tysmith) [reply FTB/Signatures/CrashInfo.py]: I added a 'strcat-param-overlap' report to the test set. I don't think there are any other variations the require a test. From what I can find they all match "\w+-param-overlap:".

**Pull request merged**